### ISSUE TITLE
zencache\cdn_filters::filter_url() Requires PHP v5.4+

### ISSUE THREAD

**Issue by jaswrks** [wpsharks/comet-cache#426]
I [got a report](https://websharks.zendesk.com/agent/tickets/5367) about Static CDN Filters not running as expected on PHP v5.3, due to a protected method that is called from an anonymous/unsupported context in PHP v5.3. That's news to me!

```
Call to protected method zencache\cdn_filters::filter_url() from context ''
```

So for the moment, it would _appear_ that Static CDN Filters require PHP v5.4+.

---

**Suggested change:** Make [this](https://github.com/websharks/zencache-pro/blob/000000-dev/zencache-pro/includes/cdn-filters.php#L303) a public method in the next release of ZenCache Pro.

**Comment by raamdev** [comment]
I'm labeling this as a bug since the Static CDN Filters feature is currently broken with PHP v5.3.

**Comment by jaswrks** [comment]
I've had this corrected. I'll submit a Pull Request once the repo changeover is done.

**Comment by skipthedrive** [comment]
Thanks for looking into this, guys!

**Comment by skipthedrive** [comment]
Hey Guys -

I wanted to give you an updated. I called my hosting provider and had them upgrade my PHP version to 5.4. I verified (with one of my plugins) that the version I am running is 5.4.19. I then enabled CDN using the default settings (no exclusions/inclusions), and added my CDN host name. I got the same error message.

Pete

**Comment by jaswrks** [comment]
> got the same error message.

Thank you for the feedback @skipthedrive ~ I thought it was strange. I'm taking a closer look at this so I can see exactly which version is in conflict here.

**Comment by skipthedrive** [comment]
No problem @jaswinc . Would SSL conflict with the CDN in any way?

**Comment by jaswrks** [comment]
My detailed tests here demonstrate that PHP v5.4.0 is all that's required to get this working. Notice that PHP versions 5.4.0+ all produce the expected `hello` output here: http://3v4l.org/8HLW5

> I verified (with one of my plugins) that the version I am running is 5.4.19. I then enabled CDN using the default settings (no exclusions/inclusions), and added my CDN host name.

Interesting. That seems not possible based on the tests that I just ran. I'll leave this open.

**Comment by skipthedrive** [comment]
OK, thanks for the update.

**Comment by jaswrks** [comment]
@raamdev Here is a version of the same test that succeeds on PHP v5.3+. Notice here that I changed the protected method to a public method. That's what the related PR does too.

http://3v4l.org/J7gnO

**Comment by raamdev** [comment]
@jaswsinc Thank you!

This has been merged into the development trunk and will go out with the next release:
- **Bug Fix** (Pro): The Static CDN Filters feature was calling a protected method that was requiring PHP v5.4+ but now the Static CDN Filters feature works with PHP v5.3+. See [#426](https://github.com/websharks/zencache/issues/426).

**Comment by raamdev** [comment]
This has been fixed in the development trunk and the next release of ZenCache will include this fix.
- **Bug Fix** (Pro): The Static CDN Filters feature was calling a protected method that was requiring PHP v5.4+ but now the Static CDN Filters feature works with PHP v5.3+. See [#426](https://github.com/websharks/zencache/issues/426).

**Comment by skipthedrive** [comment]
Thanks for the update!
